Create video

henpe071 opened the pull request:
Create a video from intervals.

In the context menu of interval items the items can be added to the Videoedit widget. Here the intervals can be edited in length and removed. Finally a video can be created from the intervals. In a dialog the user can select which size the video should be. The selectable options include all the different sizes of the included intervals and the largest width and height. There also is an option to keep all sizes the same as they were. The video will then be the largest width and height but hte videos won't be scaled and the rest of the video will be fillled with black.

Can also select the fps from the different frame rates of the video project the intervals came from.

Currently the video will be an .mkv file and with the codec mjpg and saved in the project folder under GeneratedVideos.

EDIT: 
Added Progress bar and moved it to another thred to prevent the giu from locking up.

The interval items icon is now from the first frame in the interval.

Added a new button to directly add an interval to the widget without having to create an interval-tag. shortcut 'p' currently.

Doubleclicking an interval in the widget will show that video and jump to the first frame of that interval. This only works if the videoproject can be found in the treewidget.

when generating a video from the videoedit widget, the newly made video will be added to the projecttreewidget in the folder "Generated videos". This folder is created if it does not already exist.

nisslas73 committed:
first commit

nisslas73 committed:
GUI components finished

nisslas73 committed:
something

nisslas73 committed:
context menu added

nisslas73 committed:
temporary icons

henpe071 committed:
Merge branch 'video-edit' into create-video

henpe071 committed:
Made new branch for the video creator

henpe071 committed:
Can now add interval to videoedit widget

can also edit the intervals in the widget

henpe071 committed:
Can generate video but need to fix some scaling problems

henpe071 committed:
Fix problem with the sizes when generating video

Also added an option to keep all sizes and fill the rest with black

henpe071 committed:
Can now remove items from the widget

henpe071 committed:
Generated videos are now created in the project folder

under "GeneratedVideos"

henpe071 committed:
clear widget when changing project

henpe071 committed:
Intervalicon is now first frame of interval

henpe071 committed:
Added a progressbar and a abort button

henpe071 committed:
Added a new button to directly add an interval to the video edit widget

henpe071 committed:
double clicking in widget to open the video at the first frame

henpe071 committed:
The generated video will be added to the tree

It's added under the generated videos folder that's created if it doesn't already exist

niklasniklas commented:
Works fine. 
I think I've solved the multiple-line problem. Show you tomorrow...

henpe071 committed:
Small fixes

Fix dialog when wrong path for video.
Fix when saving temp pproj with temp path update the newly generated videos path.

henpe071 committed:
Save videoedit widget but not yet sort

the operator...

henpe071 committed:
Edit widget is now sorted correctly

Also changed bookmarksort. Bookmarkitem and category now are derived from bookmarkbase

henpe071 committed:
Add name to items

niklasniklas commented:
- [x] * I get a crasch when saving from a temoprary project, with clips in video-edit widget
- [x] * Does not rotate or flip a video when generating video
- [x] * Default should be: Keep all sizes
- [ ] * Is it possible to have the text below the icon in horizontal view? (wait with this, not prio)

niklasniklas commented:
Feature request: Add a title screen. Only allowed to put in text in a "Text edit"-widget with 3 lines. Generates a black screen for 5 sek duration with the specified text.

niklasniklas commented:
- [x] Feature request: User specify resolution to any setting

henpe071 committed:
Merge branch 'development' into create-video

henpe071 committed:
Changed map for the recent file and added a new data structure for the videoedit interval to fix a crash when reloading

henpe071 committed:
Added customsizes when generating a video

henpe071 committed:
Rotation, flip and color corrections now saves and are considered when generating the video

niklasniklas commented:
- [x] Counting in progressbar should start at 1 (not 0)

- [x] I get a crash when unclicking "Keep default resolutions" (Unable to reproduce)

- [x] When generating a video with a temporary project, the path to the generated videos in the project tree does not get updated. Test with more than one video generated.

- [x] Reordering clips when saving first time (from temp-prj-file). I created 2 clips from video 1: 1A and 1B. I created 2 clips from video 2: 2A and 2B.  Before saving the first time I placed them in the order: 1A, 2A, 1B, 2B. After save is complete they com in order: 1A, 1B, 2A, 2B. This works when re-saving the project.

henpe071 committed:
Fixes some bugs with sequences and temp proj

niklasniklas commented:
Do them in the number order:
### Feature request:
2. - [x] Feature request: Add a title screen. #
Add this to "New video -dialog", with 
- a checkbox "Add title screen"
- two lines of textedit (limited in length

Produces a black screen with white text (same font as used for annotation), duration 5 sek.

3. - [x] Feature request: Custom frame rate #

### Image sequence:

1. - [ ] Does not flip, rotate or get the brighness, etc.

- [x] Name of image seq imported as video and mad into a clip in video edit is wrong. Should be the name of the sequence (not  %010d)

4. - [ ] Double click on img seq imported as video does not work

henpe071 committed:
Added custom fps and title screen

niklasniklas commented:
- [x] Interval menu: Does not have "{ - I" or "} - O" #
- [x] export menu: Change to: "Export interval to stills..." #
- [x] Menu Help: Open a pdf-document 
- [x] Edit/Options...: change to:
- "Line thickness"
- "Frame step"
- "Floating video widget (max)"

henpe071 committed:
Updated the menus

niklasniklas commented:
In dialog "New Video"

- Move "Keep default resolution" to line no.2 

- Add a second row of text to title screen

- Add duration for title screen

henpe071 committed:
Added more customization to the createvideodialog

henpe071 committed:
Open pdf

Problem with program finding file in resource

henpe071 committed:
Updated options

niklasniklas commented:
When I have a long Title in new video it does not fit:'
ex: "5000-K12345-17" or "afsdjfjsdgkljsdgjsdgsksdgljs"
When I tried "Sundsvall" it worked fine.

Did not get the "open help-pdf" to work. (Some path issue I guess) Do I have to start the program from explorer for it to work?

henpe071 committed:
Scale title text to always fit on video

Also fix help pdf

henpe071 committed:
Sequence now rotates

henpe071 committed:
Merge branch 'development' into create-video
